### PR TITLE
pl_orders mappings + move reqs into separate domain folder

### PR DESCRIPTION
**Pull request by iaincrawf**
still need to map a couple of new classes, temp placeholders. Lots of todos, not sure if they should be put into main?
Mostly making this pr since we're off for long weekend

### CODE REVIEW

**Review by richardiphillips — APPROVED** [review]
my brain only has remaining capacity for a skim but looks reasonable

File: src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDelivery.cs [hunk 1]
```diff
@@ -31,5 +33,21 @@ public class PurchaseOrderDelivery
         public DateTime? CallOffDate { get; set; }
 
         public decimal? BaseOurUnitPrice { get; set; }
+
+        public string SupplierConfirmationComment { get; set; }
+
+        public decimal? OurUnitPriceCurrency { get; set; } //our price
+
+        public decimal? OrderUnitPriceCurrency { get; set; } //order_price
```
Comment (lewisrenfrew) [inline src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDelivery.cs:41]: these comments could go

File: src/Domain.LinnApps/PurchaseOrders/CancelledPODetail.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        //REASON_CANCELLED NOT NULL VARCHAR2(600)
+        //PERIOD_CANCELLED NUMBER
+        //PERIOD_FIL_CANCELLED NUMBER
+        //VALUE_CANCELLED NUMBER(14,2)
+        //DATE_UNCANCELLED DATE
+        //DATE_UNFIL_CANCELLED DATE
+        //DATE_PREVIOUSLY_CANCELLED DATE
+        //DATE_PREVIOUSLY_FIL_CANCELLED DATE
+        //VALUE_FIL_CANCELLED NUMBER(14,2)
+        //BASE_VALUE_FIL_CANCELLED NUMBER(14,2)
```
Comment (lewisrenfrew) [inline src/Domain.LinnApps/PurchaseOrders/CancelledPODetail.cs:24]: might as well just type these in c#

File: src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs [hunk 3]
```diff
@@ -29,5 +31,46 @@ public class PurchaseOrderDetail
...
+
+        public string OrderUnitOfMeasure { get; set; }
+
+        public decimal? Duty { get; set; }
+
+        public decimal? OurUnitPriceCurrency { get; set; } //our price
+        //from mini order trigger:
+        //next_our_unit_price = :new.our_price,
+        //next_order_unit_price = :new.order_price,
+        public decimal? OrderUnitPriceCurrency { get; set; } //order  price
```
Comment (lewisrenfrew) [inline src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs:49]: comments again

File: src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs [hunk 4]
```diff
@@ -29,5 +31,46 @@ public class PurchaseOrderDetail
...
+        public decimal? BaseDetailTotal { get; set; }
+
+        public string DeliveryInstructions { get; set; }
+
+        public Employee DeliveryConfirmedBy { get; set; }
+
+        public int DeliveryConfirmedById { get; set; }
+
+        //todo map and uncomment these
+        //public CancelledPODetail CancelledDetail { get; set; }
```
Comment (lewisrenfrew) [inline src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs:70]: might as well just do this todo if its a two second job

**Review by lewisrenfrew — APPROVED** [review]
i'd personally not commit todo-y comments to main.. but not too bothered

File: src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs (hunk 4 above)
Comment (iaincrawf) [reply src/Domain.LinnApps/PurchaseOrders/PurchaseOrderDetail.cs:70]: Yeah, brain was fried and I wanted to nip outside, but wanted to get this up before everyone finished for the day so I left some bits. Turned out there were actually less todos left than I thought

**Pull request merged**